### ISSUE TITLE
Trip service with source option set to first returns InvalidQuery

### ISSUE THREAD

**Issue by narzero** [Project-OSRM/osrm-backend#3695]
I have a trip where the first source should be the starting and end point. When looking at the documentation https://github.com/Project-OSRM/osrm-backend/blob/master/docs/http.md#trip-service I see that `roundtrip` is set to `true` by default. So the only thing I need to do is specify the source of the trip since it's default value is set to `any`.

When doing:

```
curl '108.61.166.100:5000/trip/v1/driving/4.4898618,51.9854004;4.4999844,51.9881537;4.484153399999999,51.9893977;4.4521652,51.9835727;4.504290999999999,51.9840246;4.4999844,51.9881537?source=first'
```

It returns:
```
{"message":"Query string malformed close to position 159","code":"InvalidQuery"}
```

When I remove `?source=first` the call works fine.

I'm using version 5.5.4.

**Comment by daniel-j-h** [comment]
You're looking at the docs for master but you are running 5.5.4. The feature you are searching for is in master (and will be in the next stable 5.6 release), but is not in 5.5.4. Sorry.

Here are the docs for the 5.5 branch:
https://github.com/Project-OSRM/osrm-backend/blob/5.5/docs/http.md#trip-service

You can build master from source if you need the feature _right now_, or wait a bit as 5.6 is expected to be released in the next weeks.

**Comment by narzero** [comment]
Thanks for the quick response.

In 5.5.4 all trips are treated as a roundtrip where the starting point and end point is the first location. So all I needed to do in order to get the desired result was add the start as the first coordinate is the url.

Example:
```
                                      |begin & end location|
                                       \                  /
                                        ˅˅˅˅˅˅˅˅˅˅˅˅˅˅˅˅˅˅
curl '{{REDACTED}}:5000/trip/v1/driving/4.499994,51.988153;4.484016,51.9894651;4.451404,51.984560;4.489403,51.985260;4.504153,51.983921?steps=true'
```

**Comment by danpat** [comment]
@narzero Actually, the first coordinate is not guaranteed to be the start of the returned roundtrip route in 5.5.4.  The first coordinate in the round-trip route is essentially random.  If you're seeing that happen, then you're getting lucky.

The good news is that @ghoshkaj and @chaupow fixed this in `master` and it will be included in the 5.6.0 release.  It was added here:

https://github.com/Project-OSRM/osrm-backend/commit/22186589692c03a0dd3071adce3feb5abb87f9c4#diff-0b1b7bcc1320471f0f1aa57ecae61cdaR238

**Comment by narzero** [comment]
@danpat roger that, thanks.

Looking forward to the 5.6.0 release, thanks for fixing this @ghoshkaj & @chaupow!
